Makefile

```diff
@@ -0,0 +1,157 @@
+GECKO_RONIN_COMMIT=4b1114685cac9b758e923071eff5bf4350400ebd
+GAIA_RONIN_COMMIT=64cf0c8cf0cf2bf15f6fef5a8e6a1eeb40e18c95
+# Nice attempt but doesn't work sadly...
+git_has_commit = $(shell cd $(1) && git merge-base --is-ancestor $(2) HEAD && exit 0 || exit 1)
+
+##### Default make goal
+.DEFAULT_GOAL=build
+
+
+help:
+	@echo "Usage: make [command]"
+	@echo ""
+	@echo "Useful commands:"
+	@echo ""
+	@echo "\tbuild: builds Ronin (default command)"
+	@echo "\trun: run Ronin in a nested X server"
+	@echo "\tpackage: builds an installable .deb package"
+	@echo "\tclean: delete build folders/files"
+	@echo ""
+	@echo "Advanced usage commands:"
+	@echo ""
+	@echo "\tbuild-gaia: build gaia"
+	@echo "\tbuild-gecko: build gecko"
+	@echo ""
+	@echo "Variables affecting build process:"
+	@echo ""
+	@echo "\tBUILD_DIR: absolute path to build output (default: $(abspath ${mkfile_dir}/../build))"
+	@echo "\tGECKO_SOURCE_DIR: gecko git repository (default: $(abspath ${mkfile_dir}/../gecko-dev))"
+	@echo "\tGAIA_SOURCE_DIR: gaia git repository (default: $(abspath ${mkfile_dir}/../gaia))"
+	@echo "\tCUSTOM_MOZCONFIG: extends supplied mozconfig with user parameters (e.g: |mk_add_options MOZ_MAKE_FLAGS="-j3 -s"|)"
+	@echo ""
+
+${BUILD_DIR}:
+	mkdir -p ${BUILD_DIR}
+
+##### Install build dependencies
+${BUILD_DIR}/.build_deps_ready: ${BUILD_DIR}
+	python ${mkfile_dir}/bootstrap.py --application-choice desktop
+	sudo apt-get install libgstreamer-plugins-bad1.0-dev \
+		libgstreamer-plugins-base0.10-dev libgstreamer-plugins-base1.0-dev \
+		libgstreamer1.0-dev
+	touch ${BUILD_DIR}/.build_deps_ready
+
+##### Install runtime dependencies
+${BUILD_DIR}/.runtime_deps_ready:
+	sudo apt-get install xserver-xephyr
+	touch ${BUILD_DIR}/.runtime_deps_ready
+
+##### Clone gecko source if missing
+${GECKO_SOURCE_DIR}:
+	git clone -b ronin \
+		https://github.com/Phoxygen/gecko-dev ${GECKO_SOURCE_DIR}
+
+# (Todo: check that Ronin root commit is an ancestor of the active branch)
+gecko-sources: ${GECKO_SOURCE_DIR}
+
+##### Clone gaia source if missing
+${GAIA_SOURCE_DIR}:
+	git clone -b ronin \
+		https://github.com/Phoxygen/gaia ${GAIA_SOURCE_DIR}
+
+# Todo: check that Ronin root commit is an ancestor of the active branch)
+gaia-sources: ${GAIA_SOURCE_DIR}
+
+sources: gecko-sources gaia-sources
+
+##### Build gaia profile
+# (The built profile lives in ${BUILD_DIR}/profile)
+build-gaia:
+	if [ ! -d ${BUILD_DIR}/gaia/profile ]; then mkdir -p ${BUILD_DIR}/gaia/profile; fi
+	cd ${GAIA_SOURCE_DIR} && \
+	DEVICE_DEBUG=1 GAIA_DEVICE_TYPE=phone DESKTOP_SHIMS=1 NOFTU=1 PROFILE_DIR=${BUILD_DIR}/gaia/profile make
+
+${BUILD_DIR}/gaia/profile:
+	$(MAKE) build-gaia
+
+##### Prepare gecko build
+# Copy mozconfig to source dir, and append custom_mozconfig if existing
+# Note: 
+${GECKO_SOURCE_DIR}/mozconfig: ${mkfile_dir}/mozconfig ${CUSTOM_MOZCONFIG}
+	cp -v ${mkfile_dir}/mozconfig ${GECKO_SOURCE_DIR}/mozconfig
+	@if [ -n "${CUSTOM_MOZCONFIG}" ]; then \
```

Comment:
couldn't you use ifdef here? that would be pure make syntax, and it would be clearer (imho):

```
ifdef CUSTOM_MOZCONFIG
    cat $(CUSTOM_MOZCONFIG) >> $(GECKO_SOURCE_DIR)/mozconfig
endif
```